### PR TITLE
Dockerize runtime

### PR DESCRIPTION
**Pull request by alex-jw-brooks**
This pull request:
- Adds a docker file for a caikit computer vision runtime (based on caikit nlp)
- Adds definitions for text to image task
- Adds a stub module for SDXL

### CODE REVIEW

File: caikit_computer_vision/modules/text_to_image/sdxl_stub.py
Comment (gabe-l-hart) [inline caikit_computer_vision/modules/text_to_image/sdxl_stub.py:41]: It looks like there's nothing about this class that is specific to `sdxl`, right? Seems like this class isn't actually ever doing the `diffusers` load? It's not clear to me how this is actually getting a model into memory anywhere.

File: caikit_computer_vision/modules/text_to_image/sdxl_stub.py
Comment (gabe-l-hart) [inline caikit_computer_vision/modules/text_to_image/sdxl_stub.py:67]: Ah, ok, so this seems to be _just_ a stub and not actually have anything to do with `sdxl`, right? I think we should change the name unless you plan to update it to actually support running `sdxl`. As written, I think this is more of a unit test stub, right?

**Review by gabe-l-hart — COMMENTED** [review]
A couple of thoughts/comments. Depending on the goal of this PR, I think we should at least change the name of the `SDXLStub` to be `TTIStub` or something similar

File: caikit_computer_vision/modules/text_to_image/sdxl_stub.py
Comment (alex-jw-brooks) [reply caikit_computer_vision/modules/text_to_image/sdxl_stub.py:67]: Ahh yes, sorry! This is a little disorganized - it is just a stub. I am still splitting some stuff apart a bit, but I wrote this so that software could test wiring more easily.

There *is* an SDXL block too, but it's in a separate PR that is on top of this one that I am rebasing at the moment ([here](https://github.com/caikit/caikit-computer-vision/pull/60))

File: caikit_computer_vision/data_model/text_to_image.py
Comment (alex-jw-brooks) [reply caikit_computer_vision/data_model/text_to_image.py:33]: I like this more than the current output type! Currently things are written the way they are because of the way software expects image to be formatted, which is basically wrapping encoded bytes of a compressed image. I greatly prefer this also though! For now, I would like to continue with this output format if you're alright with it, since this is what they have been testing with also, but I think it would be a good idea to add this to caikit and update the result to use this in the future.

I suspect this is the route that they will want to go too, since they had already started talking about returning a JSON object with stuff + image instead of just the encoded image 🤞

File: Dockerfile
Comment (alex-jw-brooks) [reply Dockerfile:1]: Yup! It's just adding a Dockerfile for building a runtime that software can test with 😄

File: Dockerfile
Comment (alex-jw-brooks) [reply Dockerfile:18]: Good question! I think yes, but I would like to not block this PR / the SDXL PR on it if it's alright with you, since I'm about to be OOO for a while, and I would prefer to give people an image built using stuff off of `main` if possible 😄

File: caikit_computer_vision/modules/text_to_image/sdxl_stub.py
Comment (gabe-l-hart) [reply caikit_computer_vision/modules/text_to_image/sdxl_stub.py:67]: Ok, that makes sense. I still think this should be named `tti_stub` since we eventually imagine multiple modules for `tti` and this is not `sdxl` specific

File: caikit_computer_vision/data_model/text_to_image.py
Comment (gabe-l-hart) [reply caikit_computer_vision/data_model/text_to_image.py:33]: I guess I'm a little confused on how this structure accomplishes the goal of having the output be just encoded bytes. I would expect that you would still need to call some function on the `output` field to get those bytes encoded, right? If this is returned from a `task` in `caikit.runtime`, it would return as a `json` blob or serialized proto with `output` and `producer_id` fields unless there's something in the core `Image` that would provide custom serialization?

Comment (alex-jw-brooks) [reply caikit_computer_vision/data_model/text_to_image.py:33]: Yup! That is in the data model for image, the image data model holds an export format, which by default is `png`. when you get the attribute of image data on the image backend, it makes a BytesIO object and exports it with PIL [here](https://github.com/caikit/caikit/blob/main/caikit/interfaces/vision/data_model/backends/image_pil_backend.py#L57)!

File: Dockerfile
Comment (alex-jw-brooks) [reply Dockerfile:5]: Good questions - I just checked the build container, it looks like `3.9.18`. I admittedly pretty much stole this Dockerfile from caikit nlp and did not optimize it at all - the virtual env stuff, and also stages are from that.

We do not `pyTorch.compile` anything though - I think putting them into a `build` stage would be fine!

File: caikit_computer_vision/modules/text_to_image/sdxl_stub.py
Comment (alex-jw-brooks) [reply caikit_computer_vision/modules/text_to_image/sdxl_stub.py]: Agreed! Renamed

File: Dockerfile
Comment (alex-jw-brooks) [reply Dockerfile:5]: Actually, I have temporary disabled (kind of) the things that need gcc and git, so I'm just going to remove them for now - in the future, I'll reenable it in the build stage.

As far as the python venv / system python, I checked that also! Inside of the virtual environment in the container:
```
lrwxrwxrwx 1 root root   15 Jun 25 20:05 python -> /usr/bin/python
lrwxrwxrwx 1 root root    6 Jun 25 20:05 python3 -> python
lrwxrwxrwx 1 root root    6 Jun 25 20:05 python3.9 -> python
```

So yes, it looks like it is pointing at the system version, which in this base image is 3.9

**Review by gabe-l-hart — APPROVED** [review]
Ship it (once you've got the rename of `CaptionedImage` done 😉)

**Pull request merged**
